code/implementation/docker/common/src/main/java/io/cattle/platform/docker/transform/DockerTransformerImpl.java

```diff
@@ -388,6 +390,38 @@ void setLogConfig(Instance instance, Map<String, Object> fromInspect) {
 
         setField(instance, FIELD_LOG_CONFIG, logConfig);
     }
+    
+    @SuppressWarnings("unchecked")
+    void setUlimit(Instance instance, Map<String, Object> fromInspect) {
+        Object ulimits = CollectionUtils.getNestedValue(fromInspect, HOST_CONFIG, "Ulimits");
+
+        if (ulimits == null) {
+            return;
+        }
+
+        List<Ulimit> ret = new ArrayList<>();
+        if (ulimits instanceof List) {
+            for (Object ulimit : (List<Object>) ulimits) {
+                if (ulimit instanceof Map) {
+                    Ulimit l = new Ulimit();
+                    Map<String, Object> temp = (Map<String, Object>) ulimit;
+                    if (temp.get("Name") instanceof String) {
+                        l.setName(temp.get("Name").toString());
+                    }
+                    if (temp.get("Hard") instanceof Integer) {
+                        l.setHard((Integer) temp.get("Hard"));
+                    }
+                    if (temp.get("Soft") instanceof Integer) {
+                        l.setSoft((Integer) temp.get("Soft"));
+                    }
+                    if (!l.getName().isEmpty()) {
+                        ret.add(l);
+                    }
+                }
+            }
+        }
+        setField(instance, FIELD_ULIMITS, ret);
```

Comment:
I think you should only set this field if ret is not empty. This is just my preference for not setting fields on the returned instance when they are empty lists.